tests/playwright/utils/deploy_utils.py

```diff
@@ -140,26 +153,45 @@ def deploy_app(
 
     run_on_ci = os.environ.get("CI", "False") == "true"
     repo = os.environ.get("GITHUB_REPOSITORY", "unknown")
-    branch_name = os.environ.get("GITHUB_HEAD_REF", "unknown")
 
-    if (
-        not run_on_ci
-        or repo != "posit-dev/py-shiny"
-        or not (branch_name.startswith("deploy") or branch_name == "main")
-    ):
-        pytest.skip("Not on CI or posit-dev/py-shiny repo or deploy* or main branch")
-        raise RuntimeError()
+    if not (run_on_ci and repo == "posit-dev/py-shiny"):
+        pytest.skip("Not on CI and within posit-dev/py-shiny repo")
 
     app_dir = os.path.dirname(app_file_path)
-    write_requirements_txt(app_dir)
-
-    deployment_function = {
-        "connect": quiet_deploy_to_connect,
-        "shinyapps": quiet_deploy_to_shinyapps,
-    }[location]
-
-    url = deployment_function(app_name, app_dir)
-    return url
+    app_dir_name = os.path.basename(app_dir)
+
+    # Use temporary directory to avoid modifying the original app directory
+    # This allows us to run tests in parallel when deploying apps both modify the same rsconnect config file
+    with tempfile.TemporaryDirectory("deploy_app") as tmpdir:
+
+        # Creating a dir with same name instead of tmp to avoid issues
+        # when deploying app to shinyapps.io using rsconnect package
+        # since the rsconnect/*.json file needs the app_dir name to be same
+        tmp_app_dir = os.path.join(tmpdir, app_dir_name)
+        os.mkdir(tmp_app_dir)
+        shutil.copytree(app_dir, tmp_app_dir, dirs_exist_ok=True)
+        write_requirements_txt(tmp_app_dir)
+
+        deployment_function = {
+            "connect": quiet_deploy_to_connect,
+            "shinyapps": quiet_deploy_to_shinyapps,
+        }[location]
+
+        pre_deployment_time = time.time()
+        url = deployment_function(app_name, tmp_app_dir)
+        tmp_rsconnect_dir = os.path.join(
+            tmp_app_dir, "rsconnect-python", f"{os.path.basename(tmp_app_dir)}.json"
+        )
+        assert_rsconnect_file_updated(tmp_rsconnect_dir, pre_deployment_time)
+        local_rsconnect_dir = os.path.join(
+            app_dir, "rsconnect-python", f"{os.path.basename(app_dir)}.json"
+        )
+        # Copy file back if it doesn't exist locally (Helpful for local development and deployment)
+        if not os.path.exists(local_rsconnect_dir):
+
+            shutil.copy(tmp_rsconnect_dir, local_rsconnect_dir)
```

Comment:
I don't believe this can be reach via local execution. So we should remove it. My fault on the suggestion.

```suggestion
```